src/Controllers/LoginSessionController.php

```diff
@@ -29,50 +33,52 @@ class LoginSessionController extends Controller
      */
     public function remove(HTTPRequest $request): HTTPResponse
     {
-        return $this->removeLoginSession($request);
-    }
-
-    private function removeLoginSession(HTTPRequest $request): HTTPResponse
-    {
-        $failureMessage = _t(__CLASS__ . '.REMOVE_FAILURE', 'Something went wrong.');
-        try {
-            // Ensure CSRF protection
-            if (!SecurityToken::inst()->checkRequest($request)) {
-                return $this->jsonResponse([
-                    'success' => false,
-                    'message' => $failureMessage
-                ]);
-            }
+        if (empty(Security::getCurrentUser())) {
+            return $this->jsonResponse(
+                [
+                    'message' => _t(
+                        __CLASS__ . '.YOUR_SESSION_HAS_EXPIRED',
+                        'Your session has expired'
+                    )
+                ],
+                401
+            );
+        }
 
-            $id = $request->param('ID');
-            $loginSession = LoginSession::get()->byID($id);
-            if (!$loginSession) {
-                return $this->jsonResponse([
-                    'success' => false,
-                    'message' => $failureMessage
-                ]);
-            }
+        // Validate CSRF token and ID parameter
+        $id = $request->param('ID');
+        if (!SecurityToken::inst()->checkRequest($request) || empty($id) || !is_numeric($id)) {
```

Comment:
don't need the `empty($id)` check as both `!is_numeric('')` and `!is_numeric(null)` will both return true

Reply:
Fixed.